addons/sale/controllers/portal.py

```diff
@@ -140,20 +140,25 @@ def portal_order_page(
             if session_obj_date != today:
                 # store the date as a string in the session to allow serialization
                 request.session['view_quote_%s' % order_sudo.id] = today
+                subtype_id = request.env.ref('sale.mt_order_viewed')
+                follower_salesman = order_sudo.message_follower_ids.filtered(
+                    lambda follower: follower.partner_id.ids == order_sudo.user_id.partner_id.ids)
+
+                if subtype_id.id in follower_salesman.subtype_ids.ids:
```

Comment:
```suggestion
                order_user_subtypes = order_sudo.message_follower_ids.filtered(
                    lambda follower: follower.partner_id == order_sudo.user_id.partner_id
                ).subtype_ids
                if subtype_id in order_user_subtypes:
```
simplified a bit and no need to use ids where we can check recordset